pkg/apis/extensions/validation/validation.go

```diff
@@ -761,6 +762,26 @@ func validatePSPRunAsUser(fldPath *field.Path, runAsUser *extensions.RunAsUserSt
 	return allErrs
 }
 
+// validatePSPRunAsGroup validates the RunAsGroup fields of PodSecurityPolicy.
+func validatePSPRunAsGroup(fldPath *field.Path, runAsGroup *extensions.RunAsGroupStrategyOptions) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// ensure the group strategy has a valid rule
+	supportedRunAsGroupRules := sets.NewString(string(extensions.RunAsGroupStrategyMustRunAs),
+		string(extensions.RunAsGroupStrategyMustRunAsNonRoot),
+		string(extensions.RunAsGroupStrategyRunAsAny))
+	if !supportedRunAsGroupRules.Has(string(runAsGroup.Rule)) {
+		allErrs = append(allErrs, field.NotSupported(fldPath.Child("rule"), runAsGroup.Rule, supportedRunAsGroupRules.List()))
+	}
+
+	// validate range settings
```

Comment:
is it valid to specify ranges with RunAsGroupStrategyMustRunAsNonRoot or RunAsGroupStrategyRunAsAny?

Reply:
hmm , its not valid i think. The check is missing for  RunAsUser as well. I will add a check if Ranges is not nil, the strategy must be MustRunAs . Adding @pmor and @pweil- in case i am missing something

Reply:
`non-root` and `any` do not support ranges for UID at this point  cc @pmorie

Reply:
@krmayankk sounds like we need to validate that you don't get a range with a policy that doesn't support it.

Reply:
added the check